sdk/core/Azure.Core/tests/ClientDiagnosticsTests.Net50.cs

```diff
@@ -369,6 +369,54 @@ public void StartActivitySourceActivityIgnoresInvalidLinkParent()
 
             Assert.AreEqual(0, activityListener.Activities.Single().Links.Count());
         }
+
+        [Test]
+        [NonParallelizable]
+        public void ParentIdCanBeSetActivitySource()
+        {
+            using var _ = SetAppConfigSwitch();
+            string parentId = "00-6e76af18746bae4eadc3581338bbe8b1-2899ebfdbdce904b-00";
```

Comment:
Interestingly, the ID must be W3C format when calling ActivitySource.StartActivity, but not when calling SetParentId on an already started activity that was created via ActivitySource. I don't think this is an issue as I believe the ID would always be W3C format in the activity source path.

Reply:
We discovered Activity does set the parent, but we will prevent this in DiagnosticScope.
Yes, the DiagSource path has a test that sets the parent not using W3C format.